src/contracts/announcement.ts

```diff
@@ -1,63 +1,28 @@
 import Web3 from "web3";
-import { Contract } from "web3-eth-contract";
+import { MissingContract } from "../utilities/errors";
 import { AbiItem } from "web3-utils";
 
 import { getConfig, Config } from "../config/config";
 import { HexString } from "../types/Strings";
 import { MissingAccountAddress, MissingProvider } from "../utilities/errors";
 import { hashPrefix } from "../utilities/hash";
 import { TransactionReceipt } from "web3-core/types";
-
-const BATCH_CONTRACT_ADDRESS = String(process.env.BATCH_CONTRACT_ADDRESS);
-const BATCH_CONTRACT_ABI: AbiItem[] = [
-  {
-    anonymous: false,
-    inputs: [
-      {
-        indexed: false,
-        internalType: "bytes32",
-        name: "hash",
-        type: "bytes32",
-      },
-      {
-        indexed: false,
-        internalType: "string",
-        name: "dsnpUri",
-        type: "string",
-      },
-    ],
-    name: "DSNPBatch",
-    type: "event",
-  },
-  {
-    inputs: [
-      {
-        internalType: "bytes32",
-        name: "hash",
-        type: "bytes32",
-      },
-      {
-        internalType: "string",
-        name: "dsnpUri",
-        type: "string",
-      },
-    ],
-    name: "batch",
-    outputs: [],
-    stateMutability: "nonpayable",
-    type: "function",
-  },
-];
+import { getContractAddress } from "./contract";
+import { abi as announcerABI } from "@unfinishedlabs/contracts/abi/Announcer.json";
+import { Announcer } from "../types/typechain/Announcer";
 
 const GAS_LIMIT_BUFFER = 1000;
-const contract: Contract | null = null;
+const contract: Announcer | null = null;
+const CONTRACT_NAME = "Announcer";
 
-const getContract = (web3Instance: Web3): Contract => {
+const getContract = async (web3Instance: Web3): Promise<Announcer | null> => {
   if (contract) return contract;
-  return new web3Instance.eth.Contract(BATCH_CONTRACT_ABI, BATCH_CONTRACT_ADDRESS);
+  const contractAddr = await getContractAddress(web3Instance, CONTRACT_NAME);
+  if (!contractAddr) return null;
```

Comment:
If we threw an error here, we could simplify the return type of the function to just `Promise<Announcer>`. No strong opinions though.